lab-lindsay/app/entry.js

```diff
@@ -0,0 +1,62 @@
+'use strict'
+
+require('./scss/reset.scss')
+require('./scss/main.scss')
+
+const angular = require('angular')
+const cowsay = require('cowsay-browser')
+const cowsayApp = angular.module('cowsayApp', [])
+
+cowsayApp.controller('CowsayController', ['$log', CowsayController])
+
+function CowsayController($log) {
+  $log.debug('#CowsayController')
+
+  $log.log('check this out', this)
+
+  this.title = 'Welcome to Cowville'
+  this.history = []
+
+  cowsay.list((err, cows) => {
+    this.cowfiles = cows
+    this.current = this.cowfiles[0]
+  })
+
+  this.update = function(input) {
+    $log.debug('#update')
+    return cowsay.say({text: input || 'moooo', f: this.current})
+  }
+
+  this.speak = function(input) {
+    $log.debug('#speak')
+    this.spoken = this.update(input)
+    this.history.push(this.spoken)
+  }
+
+  this.undo = function() {
```

Comment:
how would you refactor this so you don't have to double click undo for it to actually undo?